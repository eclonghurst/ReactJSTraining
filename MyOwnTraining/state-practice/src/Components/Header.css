ul {
    display: flex;
    justify-content: space-around;
    align-items: center;
    list-style-type: none;
    margin: 0;
    padding: 0;
    background-color: pink;
    color: white;
  }

li {
    padding: 10px 0px 10px 0px;
}