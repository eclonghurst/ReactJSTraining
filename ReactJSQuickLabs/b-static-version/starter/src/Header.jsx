import React from "react";
import logo from './Components/images/qalogo.svg';

function Header() {
  return (
    <header>
        <nav className="navbar navbar-expand-sm">
            <a href="http://www.qa.com" className="navbar-brand" target="_blank" rel="noreferrer"><img src={logo} style={{width: '100px'}}></img></a>
            <a href="/"><h1>Todo App</h1></a>
        </nav>
    </header>
  )
}

export default Header