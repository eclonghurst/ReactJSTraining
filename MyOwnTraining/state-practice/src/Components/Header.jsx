import React from 'react'
import {Link} from 'react-router-dom';
import './Header.css';

function Header() {

  return (
    <nav>
          <ul>
            <li>
              <Link to="/">Home</Link>
            </li>
            <li>
              <Link to="/products">Products</Link>
            </li>
            <li>
              <Link to="/createProduct">Add Product</Link>
            </li>
            <li>
              <Link to="/counter">Counter</Link>
            </li>
          </ul>
    </nav>
  )
}

export default Header