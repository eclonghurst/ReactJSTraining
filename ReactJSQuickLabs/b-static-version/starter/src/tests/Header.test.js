import React from "react";
import { create } from "react-test-renderer";
import Header from "../Header";

describe('Snapshot testing', () => {
    it('Header matches snapshot', () => {
        const header = create(<Header/>);
        expect(header.toJSON()).toMatchSnapshot();
    });
}) 