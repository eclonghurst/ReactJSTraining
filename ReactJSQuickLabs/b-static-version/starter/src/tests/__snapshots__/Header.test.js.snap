// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Header matches snapshot 1`] = `
<header>
  <nav
    className="navbar navbar-expand-sm"
  >
    <a
      className="navbar-brand"
      href="http://www.qa.com"
      rel="noreferrer"
      target="_blank"
    >
      <img
        src="qalogo.svg"
        style={
          Object {
            "width": "100px",
          }
        }
      />
    </a>
    <a
      href="/"
    >
      <h1>
        Todo App
      </h1>
    </a>
  </nav>
</header>
`;
